File: baby_assistant_server/src/main/java/com/mimo/babyassistantserver/entity/KnowledgeArticle.java
package com.mimo.babyassistantserver.entity;

import java.time.Instant;
import java.util.UUID;

/** A reviewed knowledge article used as grounded context for the AI assistant. */
public class KnowledgeArticle {
    private UUID id;
    private String slug;
    private String title;
    private String category;
    private int minAgeMonths;
    private Integer maxAgeMonths;
    private String keywords;
    private String content;
    private String sourceName;
    private String sourceUrl;
    private Instant createdAt;

    public UUID getId() { return id; }
    public void setId(UUID id) { this.id = id; }
    public String getSlug() { return slug; }
    public void setSlug(String slug) { this.slug = slug; }
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    public String getCategory() { return category; }
    public void setCategory(String category) { this.category = category; }
    public int getMinAgeMonths() { return minAgeMonths; }
    public void setMinAgeMonths(int minAgeMonths) { this.minAgeMonths = minAgeMonths; }
    public Integer getMaxAgeMonths() { return maxAgeMonths; }
    public void setMaxAgeMonths(Integer maxAgeMonths) { this.maxAgeMonths = maxAgeMonths; }
    public String getKeywords() { return keywords; }
    public void setKeywords(String keywords) { this.keywords = keywords; }
    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }
    public String getSourceName() { return sourceName; }
    public void setSourceName(String sourceName) { this.sourceName = sourceName; }
    public String getSourceUrl() { return sourceUrl; }
    public void setSourceUrl(String sourceUrl) { this.sourceUrl = sourceUrl; }
    public Instant getCreatedAt() { return createdAt; }
    public void setCreatedAt(Instant createdAt) { this.createdAt = createdAt; }
}

File: baby_assistant_app/lib/models/knowledge_article.dart
class KnowledgeArticle {
  const KnowledgeArticle({
    required this.slug,
    required this.title,
    required this.category,
    required this.minAgeMonths,
    required this.maxAgeMonths,
    required this.content,
    required this.sourceName,
    required this.sourceUrl,
  });
  final String slug, title, category, content, sourceName, sourceUrl;
  final int minAgeMonths;
  final int? maxAgeMonths;
  factory KnowledgeArticle.fromApiJson(Map<String, dynamic> json) =>
      KnowledgeArticle(
        slug: json['slug'] as String,
        title: json['title'] as String,
        category: json['category'] as String,
        minAgeMonths: json['minAgeMonths'] as int,
        maxAgeMonths: json['maxAgeMonths'] as int?,
        content: json['content'] as String,
        sourceName: json['sourceName'] as String,
        sourceUrl: json['sourceUrl'] as String,
      );
  String get ageLabel => maxAgeMonths == null
      ? '$minAgeMonths 月龄起'
      : '$minAgeMonths–$maxAgeMonths 月龄';
}
